### PR TITLE
internal/testplanet

### PR DESCRIPTION
**Pull request by egonelbre**
Implement full setup intended for using in tests.

### CODE REVIEW

File: internal/testplanet/node.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		return utils.CombineErrors(err, routing.Close())
+	}
+
+	node.Kademlia = kad
+	node.Overlay = overlay.NewOverlayCache(teststore.New(), node.Kademlia)
+
+	return nil
+}
+
+// tryClose tries to guess the closing method and stop/close/shutdown the dependency
```
Comment (jtolds) [inline internal/testplanet/node.go]: ?

can we just make sure everything has a Close method and just do io.Closer?

File: internal/testplanet/planet.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+type Planet struct {
+	directory string // TODO: ensure that everything is in-memory to speed things up
+
+	nodeInfos []pb.Node
+	nodeLinks []string
+
+	nodes        []*Node
+	Satellites   []*Node // DO NOT MODIFY
+	StorageNodes []*Node // DO NOT MODIFY
+	Uplinks      []*Node // DO NOT MODIFY
```
Comment (jtolds) [inline internal/testplanet/planet.go]: how come?

Comment (egonelbre) [reply internal/testplanet/planet.go]: Ah, previously modifying the slice would've caused memory leaks. The comments were stale.

File: internal/testplanet/node.go (hunk 1 above)
Comment (egonelbre) [reply internal/testplanet/node.go]: io.Closer doesn't have context argument, which stop/shutdown sometimes need. But I did realize using a func + io.Closer would yeah get rid of this thing.

**Pull request merged**